### PR TITLE
adds a --all-files option

### PR DESCRIPTION
**Pull request by purefn**
Additionally, checks for a `.git` directory and grabs all files if it doesn't exist.

### CODE REVIEW

File: simformat.cabal [hunk 1]
```diff
@@ -1,88 +0,0 @@
-cabal-version: 1.12
```
Comment (dfithian) [inline simformat.cabal:1]: We want this file to avoid warnings

File: src/SimSpace/Config.hs [hunk 2]
```diff
@@ -28,15 +33,30 @@ newtype WhitelistFiles = WhitelistFiles { unWhitelistFiles :: [FilePath] }
...
+      srcFiles =
+        if not hasGit || allFiles
+          then Set.toList . mconcat <$> traverse listFilesRecursive ["."]
+          else lines <$> readCreateProcess (shell "git ls-files") ""
+  filter isValid <$> srcFiles
 
+listFilesRecursive :: FilePath -> IO (Set FilePath)
+listFilesRecursive dir = do
+  dirs <- listDirectory dir
+  fmap mconcat . flip traverse dirs $ \fn -> do
```
Comment (dfithian) [inline src/SimSpace/Config.hs:53]: Use for instead of flip traverse?

File: src/SimSpace/Config.hs [hunk 3]
```diff
@@ -28,15 +33,30 @@ newtype WhitelistFiles = WhitelistFiles { unWhitelistFiles :: [FilePath] }
...
       isValid fp = isSuffixOf ".hs" fp && matchesFiles fp && not (matchesWhitelist fp)
-  filter isValid . lines <$> readCreateProcess (shell "git ls-files") ""
+      srcFiles =
+        if not hasGit || allFiles
+          then Set.toList . mconcat <$> traverse listFilesRecursive ["."]
+          else lines <$> readCreateProcess (shell "git ls-files") ""
+  filter isValid <$> srcFiles
 
+listFilesRecursive :: FilePath -> IO (Set FilePath)
+listFilesRecursive dir = do
```
Comment (dfithian) [inline src/SimSpace/Config.hs:51]: I don't think this excludes .stack-work. Can you filter out hidden directories?

**Review by dfithian — CHANGES_REQUESTED** [review]

File: src/SimSpace/Config.hs (hunk 2 above)
Comment (purefn) [reply src/SimSpace/Config.hs]: Done

File: simformat.cabal (hunk 1 above)
Comment (purefn) [reply simformat.cabal:1]: D'oh. Readded.

File: src/SimSpace/Config.hs (hunk 3 above)
Comment (purefn) [reply src/SimSpace/Config.hs:52]: Done

**Review by dfithian — APPROVED** [review]

**Pull request merged**